server.js

```diff
@@ -7503,8 +7505,22 @@ cache({
             var innerText = jsonpath.join(', ');
             badgeData.text[1] = (prefix || '') + innerText + (suffix || '');
             break;
+          case 'xml':
+            data = new dom().parseFromString(data);
+            var xpathdata = xpath.select(pathExpression, data, true);
+            try {
+              badgeData.text[1] = (prefix || '') + (pathExpression.indexOf('@') + 1 ? xpathdata.value : xpathdata.firstChild.data) + (suffix || '');
+            } catch (e) {
+              switch (e.toString()){
+                case "TypeError: Cannot read property 'value' of undefined":
+                case "TypeError: Cannot read property 'firstChild' of undefined":
```

Comment:
There is not test for this case (`TypeError: Cannot read property 'firstChild' of undefined`). Would you like to add it?